### PR TITLE
adding plan and container settings to function app

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
These features are necessary to create container based function apps (like through the portal)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_functionapp

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_functionapp.py [hunk 1]
```diff
@@ -34,6 +34,29 @@
     location:
         description:
             - Valid Azure location. Defaults to location of the resource group.
+    plan:
+        description:
+            - App service plan. Required for creation.
+            - It can be name of existing app service plan in same resource group as web app.
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_functionapp.py]: ` as web app` sounds confusing..

File: test/integration/targets/azure_rm_functionapp/tasks/main.yml [hunk 2]
```diff
@@ -84,6 +84,39 @@
...
+    name: applan{{ fixed_resource_prefix }}
+    location: eastus
+    sku: S1
+    is_linux: true
+    number_of_workers: 1
+
+- name: "Create azure function app {{ function_app }}"
+  azure_rm_functionapp:
+    resource_group: "{{ resource_group }}"
+    name: af{{ fixed_resource_prefix }}container
```
Comment (yungezz) [inline test/integration/targets/azure_rm_functionapp/tasks/main.yml]: can you make it a variable?

File: lib/ansible/modules/cloud/azure/azure_rm_functionapp.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_functionapp.py]: yes, actually it's not required. if not specified, something will be created anyway by the service.

**Comment by yungezz** [comment]
ci failed

**Comment by zikalino** [comment]
@yungezz I think I have just fixed and issue with **azure_rm_appserviceplan**. hope now ci will succeed

**Comment by zikalino** [comment]
noooooooooo..... still something wrong with app service plan :-(

File: lib/ansible/modules/cloud/azure/azure_rm_functionapp.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_functionapp.py]: updated the doc

File: test/integration/targets/azure_rm_functionapp/tasks/main.yml (hunk 2 above)
Comment (zikalino) [reply test/integration/targets/azure_rm_functionapp/tasks/main.yml]: done

File: lib/ansible/modules/cloud/azure/azure_rm_appserviceplan.py [hunk 3]
```diff
@@ -340,10 +342,10 @@ def create_or_update_plan(self):
             plan_def = AppServicePlan(
                 location=self.location, app_service_plan_name=self.name, sku=sku_def, reserved=self.is_linux, tags=self.tags if self.tags else None)
 
-            poller = self.web_client.app_service_plans.create_or_update(self.resource_group, self.name, plan_def)
+            response = self.web_client.app_service_plans.create_or_update(self.resource_group, self.name, plan_def)
 
-            if isinstance(poller, AzureOperationPoller):
-                response = self.get_poller_result(poller)
+            if isinstance(response, LROPoller) or isinstance(response, AzureOperationPoller):
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_appserviceplan.py:347]: let's standardize the get poller operation. how about direct get_poller_result?

**Review by yungezz — APPROVED** [review]

**Pull request merged**